### PR TITLE
remove unused import in clientDefinitions

### PR DESCRIPTION
**Pull request by dw511214992**
fix https://github.com/Azure/autorest.typescript/issues/1293

Issue description: The generated `clientDefinitions` contains credential related import, which is not used in the generated file `clientDefinions.ts`. The unused import leads to build failure when we build it by using azure-sdk-for-js 's tsconfig.

### CODE REVIEW

**Comment by joheredi** [comment]
Would you mind adding a description about what's causing the issue and the approach you took to fix the problem please?
Without this context, it is hard to make sense of the changes. For example, the first thing I see is removing some logic to check if there is clientUrl parameters. Without context I'm not sure how that's related to the issue

File: src/restLevelClient/generateClientDefinition.ts [hunk 1]
```diff
@@ -116,14 +116,6 @@ export function generatePathFirstClient(model: CodeModel, project: Project) {
   });
 
   const clientName = getLanguageMetadata(model.language).name;
-  const uriParameter = getClientUriParameter();
```
Comment (joheredi) [inline src/restLevelClient/generateClientDefinition.ts:119]: How is this related to the unused import?

File: src/restLevelClient/generateClientDefinition.ts [hunk 2]
```diff
@@ -116,14 +116,6 @@ export function generatePathFirstClient(model: CodeModel, project: Project) {
...
-
-  const { addCredentials, credentialKeyHeaderName } = getAutorestOptions();
-  const credentialTypes = addCredentials ? ["TokenCredential"] : [];
-
-  if (credentialKeyHeaderName) {
-    credentialTypes.push("KeyCredential");
-  }
```
Comment (joheredi) [inline src/restLevelClient/generateClientDefinition.ts:126]: Are we no longer supporting KeyCredential?

File: src/restLevelClient/generateClientDefinition.ts (hunk 1 above)
Comment (dw511214992) [reply src/restLevelClient/generateClientDefinition.ts:119]: This variable is not used in generating `clientDefinitions`. So I delete it.

File: src/restLevelClient/generateClientDefinition.ts (hunk 2 above)
Comment (dw511214992) [reply src/restLevelClient/generateClientDefinition.ts:126]: No. I delete it in generating `clientDefinitions`, which doesn't need to import credential related packages.

**Comment by dw511214992** [comment]
> Would you mind adding a description about what's causing the issue and the approach you took to fix the problem please? Without this context, it is hard to make sense of the changes. For example, the first thing I see is removing some logic to check if there is clientUrl parameters. Without context I'm not sure how that's related to the issue

@joheredi I have added the context the the PR decription, and please review it. thanks

**Review by joheredi — APPROVED** [review]
